modules/RTC/BandwidthLimiter.js

```diff
@@ -0,0 +1,80 @@
+/* global __filename */
+
+import transform from 'sdp-transform';
+import { getLogger } from 'jitsi-meet-logger';
+
+const logger = getLogger(__filename);
+
+/**
+ * This will save a bandwidth limit (per mline) that should be used
+ * and then, when given an SDP, will enforce that limit.
+ * Note that this will affect *outgoing* bandwidth usage, but the SDP
+ * it must modify to implement that is the *remote* description
+ */
+export default class BandwidthLimiter {
+
+    /**
+     * Create a new BandwidthLimiter
+     */
+    constructor() {
+        /**
+         * Map of mline media type to a bandwidth
+         * limit (in kbps).  Any mlines present in this
+         * map will have the associated bandwidth limit
+         * (or 'null' for no limit) enforced, meaning that
+         * it will potentially overwrite a limit
+         * already set in the given sdp.  However, if an
+         * mline is NOT present in this map, any limit
+         * in the given sdp will not be touched.
+         */
+        this._bandwidthLimits = {};
+    }
+
+    /**
+     * Set a bandwidth limit for given mline.  If
+     * limitKbps is null, the limit will be removd
+     * @param mediaType the mline media type to set the
+     * bandwidth limit on
+     * @param limitKbps the bandwidth limit, in kbps
+     */
+    setBandwidthLimit(mediaType, limitKbps) {
+        this._bandwidthLimits[mediaType] = limitKbps;
+    }
+
+    /**
+     * Enforce any configured bandwidth limits (or
+     * lack thereof) in the given sdp
+     * @param sdp the session description
+     * @returns a potentially modified session description
+     * with any configured bandwidth limits set
+     */
+    enforceBandwithLimit(sdp) {
+        logger.debug('Enforcing any configured bandwidth limits');
+        const desc = transform.parse(sdp);
+
+        desc.media.forEach(mLine => {
+            const limitKbps = this._bandwidthLimits[mLine.type];
+
+            if (typeof limitKbps !== 'undefined') {
+                if (limitKbps === null) {
+                    logger.debug(
+                        `Removing bandwidth limit for mline ${mLine.type}`);
```

Comment:
it doesn't seem like the line limit of 80 is used, can you go through all the new code and comments ? I don't want to point every single one. I haven't opened this in any editor, but this line seems much longer than all others and judging by that the comments seem to be broken too early.

Reply:
fixed...somewhat begrudgingly.